osi_environment.proto

```diff
@@ -324,4 +336,163 @@ message EnvironmentalConditions
         //
         optional uint32 seconds_since_midnight = 1;
     }
+
+    //
+    // \brief Specification of a cloud layer.
+    //
+    message CloudLayer
+    {
+        // Definition of the cloud layer, i.e. cloud layer level and sky visualization settings.
+        //
+        optional CloudLayerLevel cloud_layer_level = 1;
+
+        // Defines the cloud layer levels given by observation of total cloud amount in oktas.
+        //
+        // For visual alignment please see reference[1].
+        //
+        // The total degree of coverage indicates how large the part of the sky vault is which is covered
+        // with clouds altogether. It is given in eighths, because this division is easier to estimate
+        // for the observer. 0 eighths means that there are no traces of clouds in the sky,
+        // 4 eighths means that the sky is covered with clouds up to half, 8 eighths means that the sky is
+        // completely covered with clouds and no sky blue can be recognized.
+        //
+        // For cloud determination, the observer will choose a location from which he can overlook the entire sky.
```

Comment:
Maybe line 59-367 could be removed completely; not sure if this is needed here.
In fact "...regardless of height and genus..." does not really fit to the layer-approach.